source/extensions/filters/network/thrift_proxy/router/upstream_request.cc

```diff
@@ -262,13 +268,8 @@ void UpstreamRequest::onUpstreamHostSelected(Upstream::HostDescriptionConstShare
   upstream_host_ = host;
 }
 
-void UpstreamRequest::onResetStream(ConnectionPool::PoolFailureReason reason) {
-  if (metadata_->messageType() == MessageType::Oneway) {
```

Comment:
@rgs1
I have completely removed this special handling for oneway here.
This means a small backward incompatibility since outlier detection and stat counting will be done for oneway as well. Is it ok?

Reply:
I think that's OK (though that might be a narrow view, since we don't use Oneway). 

@fishcakez thoughts?

Reply:
Im not sure this is the right behavior as I dont think we should send an TApplicationException on a oneway, or is this hidden inside ThriftProxy that local reply isnt done then?

Reply:
But to answer the specific question. I think its okay from a thrift perspective to set outlier detection failures on oneways for conenction issues as still failed to proxy the request to the endpoint.

Reply:
> Im not sure this is the right behavior as I dont think we should send an TApplicationException on a oneway, or is this hidden inside ThriftProxy that local reply isnt done then?

Yes, in case of oneway there will be nothing sent to wire even if `sendLocalReply` is called, see `ConnectionManager::sendLocalReply`.
Actually the original behavior was not consistent for oneway since in `Router::messageBegin` the failing cases sent out the exception, but here the connection was closed directly.